man/fi_efa.7.md

```diff
@@ -134,6 +134,43 @@ provider for AWS Neuron or Habana SynapseAI.
   delivered to the target buffer only once. If endpoint is not able to support
   this feature, it will return -FI_EOPNOTSUPP for the call to fi_setopt().
 
+# PROVIDER SPECIFIC DOMAIN OPS
+The efa provider exports extensions for additional operations
+that are not provided by the standard libfabric interface. These extensions
+are available via the "`fi_ext_efa.h`" header file.
+
+## Domain Operation Extension
+
+Domain operation extension is obtained by calling `fi_open_ops`
+(see [`fi_domain`(3)](fi_domain.3.html))
+```c
+int fi_open_ops(struct fid *domain, const char *name, uint64_t flags,
+    void **ops, void *context);
+```
+and requesting `FI_EFA_DOMAIN_OPS_1` in `name`. `fi_open_ops` returns `ops` as
+the pointer to the function table `fi_efa_ops_domain` defined as follows:
+
+```c
+struct fi_efa_ops_domain {
+	int (*query_mr)(struct fid_mr *mr, struct fi_efa_mr_attr *mr_attr);
+};
+```
+
+It contains the following operations
+
+### query_mr
+This op query an existing memory registration (mr), and returns the efa
+specific mr attribute which is defined as follows
+
+```c
+struct fi_efa_mr_attr {
+    uint32_t pci_bus_id;
+};
+```
+. The `pci_bus_id` represents the pci bus id of the rx (receive, remote-write etc.)
```

Comment:
This needs more description.  I'm not actually sure what this means.  I'm also not sure how to think about it without also knowing where CQs get emitted.

Reply:
@mrgolin mentioned CQ is emmited at 0 by default.. If the user doesn't know this info, I could change this to a boolean that indicates whether rx and cqe arrives in the same pcie port